spinalcordtoolbox/scripts/sct_detect_compression.py

```diff
@@ -0,0 +1,333 @@
+                                           ignore_index=True)
+
+    return compression_df
+
+
+def get_disc_slices(fname_disc):
+    """
+    Get axial slice numbers corresponding to the intervertebral discs.
+    :param fname_disc: file with disc labels
+    :return: dictionary with disc labels as keys and corresponding axial slice numbers as values
+    """
+    im_disc = Image(fname_disc).change_orientation('RPI')
+    discs = [int(disc) for disc in np.trim_zeros(np.unique(im_disc.data))]  # get unique disc labels (e.g., 3, 4, 5 etc)
+    # Find axial slices corresponding to the discs
+    # Note: [0] is used because get_slices_from_vertebral_levels returns list
+    disc_slices = {disc: get_slices_from_vertebral_levels(im_disc, disc)[0] for disc in discs}
+
+    return disc_slices
+
+
+def main(argv: Sequence[str]):
+    parser = get_parser()
+    arguments = parser.parse_args(argv)
+    verbose = arguments.v
+    set_loglevel(verbose=verbose, caller_module_name=__name__)
+
+    # Parse arguments
+    fname_seg = get_absolute_path(arguments.s)
+    fname_disc = get_absolute_path(arguments.discfile)
+    angle_correction = bool(arguments.angle_corr)
+
+    # Compute shape metrics per slice
+    metrics_agg_merged = compute_shape_metrics(fname_seg, fname_disc, angle_correction, verbose)
+
+    # Get discs corresponding to the compression
+    disc_slices = get_disc_slices(fname_disc)
+
+    # Process axial slices for each disc to compute compression probability
+    compression_df = process_compression(metrics_agg_merged, disc_slices, arguments.num_of_slices)
+
+    # Save the results to a CSV file
+    fname_out = splitext(add_suffix(fname_seg, '_compression_results'))[0] + '.csv'
+    compression_df.to_csv(fname_out, index=False)
+    printv(f"Results saved to: {fname_out}", verbose)
+
+    # Loop across discs and print results to terminal
+    for disc in compression_df['Disc'].unique():
+        # If any axial slice around the disc is compressed, the disc is considered compressed
+        if 'yes' in compression_df.loc[compression_df['Disc'] == disc, 'Compression probability category'].values:
+            printv(f"Disc {disc}, compression probability: yes", verbose)
+        elif 'possible' in compression_df.loc[compression_df['Disc'] == disc, 'Compression probability category'].values:
+            printv(f"Disc {disc}, compression probability: possible", verbose)
+        elif 'no' in compression_df.loc[compression_df['Disc'] == disc, 'Compression probability category'].values:
+            printv(f"Disc {disc}, compression probability: no", verbose)
+
+    if verbose == 2:
+        for index, row in compression_df.iterrows():
+            printv(f"Disc {(row['Disc'])}, axial slice {int(row['Axial slice #'])}: "
+                   f"probability of compression: {row['Compression probability'] * 100:.2f}% "
+                   f"(compression probability category: {row['Compression probability category']})", verbose)
```

Comment:
I think it would be nice to always have the percentage printed too!

Reply:
Good point! Done in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4760/commits/d0eb78eedf45f71fa289194faa4cf314e86684fd.